### PR TITLE
Refine json object test for NDEBUG case

### PR DESCRIPTION
**Pull request by mellery451**

### CODE REVIEW

File: src/test/json/Object_test.cpp [hunk 1]
```diff
@@ -209,13 +209,17 @@ class JsonObject_test : public ripple::test::TestOutputSuite
...
-#ifdef DEBUG
         setup ("repeating keys");
         auto& root = makeRoot();
         root.set ("foo", "bar");
         root.set ("baz", 0);
-        auto fail = [&]() { root.set ("foo", "bar"); };
-        expectException (fail);
+        // setting key again throws in !NDEBUG builds
+        auto set_again = [&]() { root.set ("foo", "bar"); };
+#ifdef NDEBUG
```
Comment (ximinez) [inline src/test/json/Object_test.cpp]: Why switch from `DEBUG` to `NDEBUG`?

Comment (mellery451) [reply src/test/json/Object_test.cpp]: the code that is being tested is predicated on `NDEBUG`:  https://github.com/ripple/rippled/blob/710d9b6538f7537cf5f61fdec207e94748e14329/src/ripple/json/impl/Writer.cpp#L150  ..so I figured we should at least align with that. Also, whereas `NDEBUG` is fairly standard (assert depends on it), `DEBUG` is not and we might want consider removing it from our macros.

Comment (ximinez) [reply src/test/json/Object_test.cpp]: I missed the symmetry when I was digging into the writer. That's an excellent point.

**Comment by nbougalis** [comment]
Merged as 5741a8356f359c6bec0cd68a95b0725fb8d9ba8b

**Pull request closed without merging**